src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp

```diff
@@ -0,0 +1,47 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstdlib>
+#include <utility>
+
+#include "Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodType.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim, typename T>
+class DirectionMap;
+template <size_t VolumeDim>
+class Element;
+template <size_t VolumeDim>
+class Mesh;
+/// \endcond
+
+namespace Minmod_detail {
+/// \brief Implements the troubled-cell indicator corresponding to the Minmod
+/// limiter.
+///
+/// The troubled-cell indicator (TCI) determines whether or not limiting is
+/// needed. See SlopeLimiters::Minmod for a full description of the Minmod
+/// limiter. Note that as an optimization, this TCI returns (by reference) some
+/// additional data that are used by the Minmod limiter in the case where the
+/// TCI returns true (i.e., the case where limiting is needed).
+template <size_t VolumeDim>
+bool troubled_cell_indicator(
```

Comment:
Is there a reasonable way to write a factored test for this?

Reply:
Yes, there is. As I wrote in the PR description: "A future PR will refactor the Minmod tests, by separating them into TCI tests and limiter tests."

The factoring isn't quite done yet, but if you prefer I can finish it up and include it as part of this PR.

Reply:
@nilsdeppe agrees that this can wait for a future PR

Reply:
Tests are factored in PR #1433